### ISSUE TITLE
Hide Warp icon from the Dock when using Hotkey window

### ISSUE THREAD

**Issue by ryarturogi** [warpdotdev/Warp#1154]
### Describe the solution you'd like?

```text
Like in xterm2, we can hide the icon from showing in the dock cause if we use the shortcuts to toggle up the terminal. There's no reason. 

It's always working in the background and in the settings there must be the option to show/hide the icon to maintain the cleanest dock we can have.
```

**Comment by xxleyi** [comment]
Maybe also hide from `cmd + tab` switcher?

**Comment by willian** [comment]
I'd like to be able to set any hotkeys, like `Ctrl+Shift+~`. At the moment, Warp does not allow this.

**Comment by jacobgraf** [comment]
I love the Hotkey window, but find myself accidentally clicking the dock icon which opens another terminal outside of the hotkey window. I would like an option to hide Warp from the dock and also be able to easily start Warp in the background at system startup. I don't want a window to open as startup, I'd only like the global hotkey to work right from a fresh boot.

Another alternative to this is, when the hotkey is enabled and you click the dock icon, have it open your hotkey session rather than creating a new session.

Both options would work, but I would prefer the former.

Thanks!

**Comment by elviskahoro** [comment]
Will circle back on this thread when there are updatese

**Comment by anfomin** [comment]
Hiding Dock icon when using Hotkey window will be great feature!

**Comment by bogdanrn** [comment]
similar to #981

**Comment by mahyess** [comment]
Is there any workaround to hide the dock icon? 

Asking because this feature doesn't seem to be a priority at this moment.

**Comment by anfomin** [comment]
> Is there any workaround to hide the dock icon?
> 
> Asking because this feature doesn't seem to be a priority at this moment.
> 
> Edit: [This workaround](https://discussions.apple.com/thread/7444141?answerId=31419529022#31419529022) works flawed, no icon in dock, but still a dot. Open for better suggestions.

Editing application contents is bad idea because after any update it will revert back. So I think there is no workaround until feature will be implemented.

**Comment by dvovney** [comment]
+1

Wish it could act like iTerm - there is nothing in dock when Global hotkey is enabled.

**Comment by avxkim** [comment]
+1 for removing it from CMD+TAB & dock

**Comment by halfbrilliant** [comment]
I wonder if there is a third party app/hack that can remove running apps from the Dock and CMD+Tab switcher? Seems like a thing that could theoretically exist, if I find it I'll report back

**Comment by izuolan** [comment]
> I wonder if there is a third party app/hack that can remove running apps from the Dock and CMD+Tab switcher? Seems like a thing that could theoretically exist, if I find it I'll report back

https://github.com/lwouis/alt-tab-macos
You may be looking for this software.😆

**Comment by frederikkempe** [comment]
Using Warp via the Hotkey window as well. Would be great to be able to hide it in the dock.

**Comment by fishuke** [comment]
Would be great to be able to hide it in the dock. up

**Comment by jacobgraf** [comment]
Any progress here?

I love the idea of having system-wide global access to Warp, but it's very confusing when your muscle-memory kicks in and you click the dock icon which launches a new/separate instance of Warp.

If we can't hide the dock icon completely, is there at least a setting to have it open the dedicated/hotkey window rather than spinning up another detached session?

**Comment by merlindru** [comment]
Want it gone too - the hotkey window is all i ever need and it takes up a slot in cmd+tab and the Dock!

**Comment by bogdanrn** [comment]
They added an update where you can change the icon if you want to, but still no option to hide it. Missed oportunity right there. Just add a "hidden" option to the list of icons and call it a day.

**Comment by jacobgraf** [comment]
Yeah, I've been waiting too long. Love Warp, but this missing feature is going to cause me to start looking around for other options. Frustrating.

**Comment by tomcwatts** [comment]
I'm going to be looking for alternatives, or going back to iTerm if they can't add this, how many years have we been waiting for this? The dock and CMD+tab icon just get in my way and offer no value because I use the hotkey.

**Comment by avxkim** [comment]
Released warp 2.0, but still no option to hide from dock 😂

**Comment by plakhin** [comment]
> Released warp 2.0, but still no option to hide from dock 😂

yep :( 
two things keeping me stick to iTerm2:
- show icon in menu bar instead of dock (and alt+tab) 
- and option+option hotkey

**Comment by ShankyJS** [comment]
Can we bring this up now that this is open source? @captainsafia offerring my help to implement!

**Comment by zachlloyd** [comment]
Amazing.  @ShankyJS let's start with specs. @oz-agent please create specs for this issue

**Comment by ShankyJS** [comment]
> Amazing.  @ShankyJS let's start with specs. @oz-agent please create specs for this issue

Awesome, looking forward to what OZ suggests for the spec

**Comment by zachlloyd** [comment]
Hrm, @oz-agent please try again to create the specs

**Comment by ShankyJS** [comment]
@zachlloyd answered on the spec issue.

**Comment by ShankyJS** [comment]
@zachlloyd gently nod

**Comment by zachlloyd** [comment]
@kevinyang372 is our expert here i think.  kevin, thoughts on the specs?

**Comment by kevinyang372** [comment]
A couple of comments on the product behavior:
1. I don't think this setting should only apply when global hotkey window is enabled. For example, we also have this request on making dock icon configurable in general: https://github.com/warpdotdev/warp/issues/5309
2. When we hide the dock icon, I think we still show a minimum icon in the status bar in the top right corner. Otherwise there is no way for user to check / bring up the window other than the global hotkey

**Comment by ShankyJS** [comment]
> A couple of comments on the product behavior:
> 1. I don't think this setting should only apply when global hotkey window is enabled. For example, we also have this request on making dock icon configurable in general: https://github.com/warpdotdev/warp/issues/5309
> 2. When we hide the dock icon, I think we still show a minimum icon in the status bar in the top right corner. Otherwise there is no way for user to check / bring up the window other than the global hotkey

I agree on number 2. The status bar might be a bigger lift for this specific spec though? Happy to ask oz bot for a spec if we could ship it as a separate PR

**Comment by kevinyang372** [comment]
I don't think either of my point will be a big lift. Do you mind scoping this out with the oz agent?

If it does turn out to be a bigger change, we can break it down into a few PRs behind a feature flag

**Comment by ShankyJS** [comment]
@oz-agent can you please build a spec for this requirement https://github.com/warpdotdev/warp/issues/1154#issuecomment-4375050991 if the change is too big we might split in multiple PR's feature flagged.

**Comment by ShankyJS** [comment]
@oz-agent I don't see mention of the number 2 comment: When we hide the dock icon, I think we still show a minimum icon in the status bar in the top right corner. Otherwise there is no way for user to check / bring up the window other than the global hotkey

Can you please spec that out as well?

**Comment by ShankyJS** [comment]
@oz-agent can you please try again to update the spec, including the requirement: `When we hide the dock icon, I think we still show a minimum icon in the status bar in the top right corner. Otherwise there is no way for user to check / bring up the window other than the global hotkey` please!

**Comment by ShankyJS** [comment]
@kevinyang372 I think the spec looks good now! It got the intention with the minimal status bar icon, happy to see how my PR looks including those changes, I can start working on it on the same PR I already had opened for this.  https://github.com/warpdotdev/warp/pull/9926

**Comment by kevinyang372** [comment]
Hey @ShankyJS Do you mind putting the specs together with the PR you have in progress? This way we could avoid doing two reviews on the same issue

**Comment by ShankyJS** [comment]
@kevinyang372 for sure, I can do that! I'll let you know once I have them merged so we can close the spec PR

**Comment by ShankyJS** [comment]
@kevinyang372 could you take a look to my [PR](https://github.com/warpdotdev/warp/pull/9926), also I think this issue needs the ready-to-implement label so I can trigger a new review from `oz`.

Thanks!

**Comment by avxkim** [comment]
i hope it will be landed very soon
